modules/core/src/main/java/org/apache/ignite/IgniteLogger.java

```diff
@@ -101,6 +103,17 @@ public default void trace(String marker, String msg) {
      */
     public void debug(String msg);
 
+    /**
+     * Logs out a debug message with an attached Throwable.
+     *
+     * @param msg Debug message.
+     * @param ex  Throwable to attach.
+     */
+    public default void debug(String msg, Throwable ex) {
```

Comment:
Let's implement this method in the heirs by analogy with 
`org.apache.ignite.IgniteLogger#warning(java.lang.String, java.lang.Throwable)`
`org.apache.ignite.IgniteLogger#error(java.lang.String, java.lang.Throwable)`